atmo/appsource/bundlesource.go

```diff
@@ -42,21 +50,25 @@ func (b *BundleSource) Start(opts options.Options) error {
 	return nil
 }
 
-// Runnables returns the Runnables for the app
-func (b *BundleSource) Runnables() []directive.Runnable {
+// Runnables returns the Runnables for the app.
+func (b *BundleSource) Runnables(identifier, version string) []directive.Runnable {
+	if !b.checkIdentifierVersion(identifier, version) {
+		return nil
+	}
+
 	b.lock.RLock()
 	defer b.lock.RUnlock()
 
 	if b.bundle == nil {
-		return []directive.Runnable{}
+		return nil
 	}
 
 	return b.bundle.Directive.Runnables
 }
 
 // FindRunnable searches for and returns the requested runnable
-// otherwise ErrRunnableNotFound
-func (b *BundleSource) FindRunnable(fqfn, auth string) (*directive.Runnable, error) {
+// otherwise ErrRunnableNotFound.
+func (b *BundleSource) FindRunnable(fqfn, _ string) (*directive.Runnable, error) {
```

Comment:
This is a bit funny. Since we're changing the overall `AppSource interface{}` can we just eliminate this since it's unused here? Can we decouple authentication from this part of the interface?

I see it's used in other parts of the codebase.

Reply:
The only place it's being used is in the httpsource implementation. If I understand correctly, you're saying that `auth` is more of an implementation level detail rather than something that the interface should care about

Reply:
Opened https://github.com/suborbital/atmo/issues/133 to track this